tests/arc/tstrformat.nim

```diff
@@ -2,9 +2,20 @@ discard """
   output: '''
 verstuff
 '''
-  cmd: "nim c --gc:arc $file"
+  matrix: "--gc:arc"
+  targets: c
+  knownIssue: '''
+    Semantic analysis duplicates the analysed argument into each usage
+    inside the template. Since it contains definitions of local variables,
+    multiple definitions of the same entity exist in the expanded code, which
+    is illegal at the MIR level, thus making the `injectdestructors` pass fail
+    '''
```

Comment:
I decided to disable this test for now. It's the most severe issue, but there are so many hacks in `mirgen` already that it felt wrong to introduce yet another one. For now, an entity being defined multiple times will raise an internal compiler error when the code reaches the `injectdestructors` pass.

Reply:
This sounds like it should be fixed in semantic analysis, rather than hack up `injectdestructors`.

My next two days are really busy, but I could peek at sem and eval template code on the weekend, not sure what the fix will entail, though

Reply:
Yep, semantic analysis is the correct place to fix this.

The issue here was what originally motivated me to look into the macro output sanitizer. Turning the typed AST passed to a `typed` template parameter into untyped AST before inserting it into the body could fix the problem, but work on the sanitizer is currently stuck due to issues with gensym'ed symbols.